src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
Yes, that is definitely a subject to discussion. Moreover, it was one main purpose of the ticket: https://techjira.zalando.net/browse/ARUHA-88.

The business Code shouldn't be aware of the HTTP stack, I agree with you. But it doesn't have to if the Exception itself provides a status code. We wanted to see how far we can push this idea.